### PR TITLE
Fix doctests of `open_pickle_dataset`

### PR DESCRIPTION
**Pull request by toslunar**

### CODE REVIEW

**Comment by toslunar** [comment]
Jenkins, test this please.

File: chainer/datasets/pickle_dataset.py [hunk 1]
```diff
@@ -145,7 +145,7 @@ def open_pickle_dataset_writer(path, protocol=pickle.HIGHEST_PROTOCOL):
     :func:`PickleDatasetWriter.close` or use `with`:
 
     >>> with chainer.datasets.open_pickle_dataset_writer('path') as writer:
-    ...    # use writer
+    ...     pass  # use writer
```
Comment (beam2d) [inline chainer/datasets/pickle_dataset.py]: Maybe the same fix is needed at L121?

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by toslunar** [comment]
Jenkins, test this please.

File: chainer/datasets/pickle_dataset.py [hunk 2]
```diff
@@ -53,11 +53,13 @@ class PickleDataset(dataset_mixin.DatasetMixin):
     Even when a user wants to use a large dataset, this dataset can stores all
     data in a large storage like HDD and each data can be randomly accessible.
 
-    >>> with chainer.datasets.open_pickle_dataset_writer('/path/to/data') as w:
+    >>> import tempfile
+    >>> _, path_to_data = tempfile.mkstemp()
```
Comment (beam2d) [inline chainer/datasets/pickle_dataset.py]: Could you put these lines into `testsetup::` and add `testcleanup::` to close the file?

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by beam2d** [comment]
LGTM

**Pull request merged**